libs/wallet/src/types.ts

```diff
@@ -0,0 +1,130 @@
+import { type StoreApi } from 'zustand';
+import {
+  type Transaction,
+  type TransactionResponse,
+} from './transaction-types';
+import { type Chain } from './chains';
+import { type ConnectorError } from './errors';
+
+export enum JsonRpcMethod {
+  ConnectWallet = 'client.connect_wallet',
+  DisconnectWallet = 'client.disconnect_wallet',
+  ListKeys = 'client.list_keys',
+  SignTransaction = 'client.sign_transaction',
+  SendTransaction = 'client.send_transaction',
+  GetChainId = 'client.get_chain_id',
+}
+
+export interface TransactionParams {
+  publicKey: string;
+  transaction: Transaction;
+  sendingMode: 'TYPE_SYNC';
+}
+
+export type VegaWalletEvent = 'client.disconnected';
+
+export type ConnectorType =
+  | 'injected'
+  | 'jsonRpc'
+  | 'snap'
+  | 'viewParty'
+  | 'mock';
+
+export interface Connector {
+  readonly id: ConnectorType;
+  readonly name: string;
+  readonly description: string;
+
+  bindStore(state: StoreApi<Store>): void;
+  connectWallet(chainId?: string): Promise<{ success: boolean }>;
+  disconnectWallet(): Promise<{ success: boolean }>;
+  getChainId(): Promise<{ chainId: string }>;
```

Comment:
What does the return value of success add here out of interest? Can this not just be `void` if we are throwing errors now?